### DISCUSSION TITLE
getting 400 Bad request on conversationalai streaming request

### DISCUSSION THREAD

**Discussion by davehorton** [deepgram/community#414] [General help] [answered]
I'm getting a 400 Bad Request when sending this path in my websocket connection
```
/v1/listen?model=conversationalai&language=en_us&interim_results=true&encoding=linear&sample_rate=8000
```

Any thoughts on what is wrong?  I am pretty sure it is the model (conversationalai) since if I try other models (e.g. nova-2-ea) it works, but not sure why that should be.

The request id was 10a69986-c7ea-4baf-b1c8-ebe2e2a18fde and performed at 11/03/2023 11:52:27

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @davehorton, I tested this myself with a local mic using our [streaming test suite](https://github.com/deepgram/streaming-test-suite/tree/main). I found that the `conversationalai` model works fine, but the other parameters require some slight modifications.

Please change the language code from `language=en_us` to `language=en-US`. Our standard format for US English is `en-US`, and lowercase `en-us` will also work, but `en_us` with an underscore will not. 

I believe you'll also want to switch from `encoding=linear` to `encoding=linear16`.

I'm using a sample rate of `16000` instead of `8000`, but either should work given the appropriate streaming setup.

Can you make these changes and see if that works for you?

Exact path I am using:
```
/v1/listen?model=conversationalai&language=en-US&interim_results=true&encoding=linear16&sample_rate=16000
```

**Comment by davehorton** [comment]
Thanks!  This fixed it.  

Also, wonder if you have an opinion on a related question.  For a conversational AI use case, is 'conversationalai' still the best model to use, or would the nova or nova-2 model actually work better?

**Reply by jkroll-deepgram** [reply]
Good question - our `conversationalai` domain model has only ever been available for our `base` tier ([see doc](https://developers.deepgram.com/docs/tier)). It is currently not a highly-used model of ours. Nova and Nova-2 tiers generally offer better accuracy and price than Base. However, we are planning to retrain `conversationalai` with the Nova-2 architecture.

I'd certainly recommend some testing on your own audio to see if you notice any difference for your specific use case, but in general, you may be better served by using Nova/Nova-2 (either the `general` model, or `phonecall` if you have lower-quality audio, such as from phonecalls or a local mic). Then when we release a Nova-2 `conversationalai` model, you could consider moving to that one.
